sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java

```diff
@@ -401,7 +400,24 @@ public ConfigurationClientBuilder pipeline(HttpPipeline pipeline) {
      * @return The updated ConfigurationClientBuilder object.
      */
     public ConfigurationClientBuilder configuration(Configuration configuration) {
-        this.configuration = configuration;
+        this.configuration = configuration.getSection("appconfiguration");
+
+        // will populate ClientOptions or HttpClientOptions depending on config
+        clientOptions(HttpClientOptions.fromConfigurationOrDefault(this.configuration, this.clientOptions));
+
+        httpLogOptions(HttpLogOptions.fromConfigurationOrDefault(this.configuration, this.httpLogOptions));
+
+        endpoint(this.configuration.get("appconfiguration.endpoint", this.endpoint));
```

Comment:
>appconfiguration.endpoint [](http://example.com/codeflow?start=41&length=25)

should appconfiguration here be an implicit prefix since section has already been set.

Reply:
this is a limitation of the current approach with regard to local (client-specific) properties. it's a work in progress and agree, this needs to be resolved before we finalize the approach here.